### DISCUSSION TITLE
pdm add relative path resolves in absolute paths in dependency

### DISCUSSION THREAD

**Discussion by jasonmellone** [pdm-project/pdm#3212] [Q&A] [answered]
I am following the example in [the docs](https://pdm-project.org/latest/usage/dependency/#local-dependencies) using:

`$ pdm add ./../foo/`

But when I then look in my pyproject.toml, I see:

`"foo @ file:///home/my_user_name/git/foo"`

But, based on the docs, I'd expect to see (something like - or at least I've convinced myself I should see):

` "foo @ file:///${PROJECT_ROOT}/../foo"`

**Comment by frostming** [comment]
What build backend are you using?

**Comment by jasonmellone** [comment]
```
$ pdm -V
PDM, version 2.19.2
```

```
$ python
Python 3.12.7 (main, Oct  1 2024, 08:52:11) [GCC 9.4.0] on linux
```

**Reply by frostming** [reply, marked as answer]
No, I mean the `[build-system]` table in `pyproject.toml`

Only `pdm.backend` supports converting relative path to `${PROJECT_ROOT}` url

**Reply by jasonmellone** [reply]
This resolved it for me
